middleware/validator/users.js

```diff
@@ -0,0 +1,72 @@
+const Validator = require('validatorjs');
+const mongoose = require("mongoose");
+const User = require('../../models/index').User;
+
+let errorMessage ='';
+
+module.exports = {
+   /*   Function validates user Signup inputs
+    with customized error messages
+   */
+   validateSignup: (req, res, next) => {
+    // User Signup inputs    
+    const userInput = {
+            username: req.body.user.username,
+            email: req.body.user.email,
+            password:req.body.user.password
+        }
+
+    // Validator rules
+    const rules = {
+        username: 'required',
+        email: 'required|email',
+        password: 'required|alpha_num|min:8'
+    }
+
+    // Setting up customized descriptive messages
+    const validation = new Validator(userInput, rules, {
+        "required.username": "You did not specify a username. Please enter a username in the specified field",
+        "required.email": "You did not specify an email. Please enter an email in the specified field",
+        "email.email": "The email you entered is invalid. Please enter a valid email",
+        "required.password": "You did not specify a password. Please enter a password in the specified field",
+        "alpha_num.password": "Your password should be alphanumeric (should contain both letters and/or numbers ONLY)",
+        "min.password": "Password entered should have minimum of 8 characters"
+    });
```

Comment:
I think the validation rules and instantiation (`const validation = new Validator()`) should be a setup outside the `validateSignup` function.